scriptmodules/emulators/mame.sh

```diff
@@ -0,0 +1,152 @@
+#!/usr/bin/env bash
+
+# This file is part of The RetroPie Project
+#
+# The RetroPie Project is the legal property of its developers, whose names are
+# too numerous to list here. Please refer to the COPYRIGHT.md file distributed with this source.
+#
+# See the LICENSE.md file at the top-level directory of this distribution and
+# at https://raw.githubusercontent.com/RetroPie/RetroPie-Setup/master/LICENSE.md
+#
+
+rp_module_id="mame"
+rp_module_desc="MAME emulator"
+rp_module_help="ROM Extensions: .zip .7z\n\nCopy your MAME roms to either $romdir/mame or\n$romdir/arcade"
+rp_module_licence="GPL2 https://github.com/mamedev/mame/blob/master/COPYING"
+rp_module_section="exp"
+rp_module_flags="!mali !armv6"
+
+function _latest_ver_mame() {
+    wget -qO- https://api.github.com/repos/mamedev/mame/releases/latest | grep -m 1 tag_name | cut -d\" -f4
+}
+
+function _get_binary_name_mame() {
+    # The MAME executable on 64-bit systems is called mame64 instead of mame. Rename it back to mame.
+    if isPlatform "64bit"; then
+        echo 'mame64'
+    else
+        echo 'mame'
+    fi
+}
+
+function depends_mame() {
+    if compareVersions $__gcc_version lt 6.0.0; then
+        md_ret_errors+=("Sorry, you need an OS with gcc 6.0 or newer to compile mame")
+        return 1
+    fi
+
+    # Additional libraries required for running
+    getDepends libsdl2-ttf-2.0-0
+
+    # Additional libraries required for compilation
+    # Note: libxi-dev is requires as of v0.210, because of flag changes for XInput
+    getDepends libfontconfig1-dev qt5-default libsdl2-ttf-dev libxinerama-dev libxi-dev
+}
+
+function sources_mame() {
+    gitPullOrClone "$md_build" https://github.com/mamedev/mame.git "$(_latest_ver_mame)"
+}
+
+function build_mame() {
+    # More memory is required for x86 platforms
+    if isPlatform "x86"; then
+        rpSwap on 8192
+    else
+        rpSwap on 4096
+    fi
```

Comment:
I suspect this would be better worked out from 64bit flag. As it's likely related to that.

Reply:
You are probably correct on this, but I do not have a 32bit linux based system to test it out on. I will make the change.

Reply:
Changed to trigger off of 64bit vs. x86.